src/storage/v2/inmemory/storage.cpp

```diff
@@ -558,6 +557,13 @@ Result<EdgeAccessor> InMemoryStorage::InMemoryAccessor::EdgeSetFrom(EdgeAccessor
         CreateAndLinkDelta(&transaction_, to_vertex, Delta::RemoveInEdgeTag(), edge_type, new_from_vertex, edge_ref);
         to_vertex->in_edges.emplace_back(edge_type, new_from_vertex, edge_ref);
 
+        if (storage_->storage_mode_ != StorageMode::ON_DISK_TRANSACTIONAL) {
```

Comment:
Do we really need this check in if? If we are here it means we are in-memory.

Reply:
We don't need this check good catch! Also removed it from a couple of lines below.